src/components/CheckEditor/FormComponents/HttpRequest.tsx

```diff
@@ -0,0 +1,246 @@
+import { GrafanaTheme2 } from '@grafana/data';
+import { Select, Stack, Tooltip, useStyles2, useTheme2 } from '@grafana/ui';
+import { css, cx } from '@emotion/css';
+
+import { HttpRequestFields, TLSConfigFields } from '../CheckEditor.types';
+import { HttpMethod } from 'types';
+import { getMethodColor, parseUrl } from 'utils';
+import { METHOD_OPTIONS } from 'components/constants';
+import { Indent } from 'components/Indent';
+import { QueryParams } from 'components/QueryParams';
+import { Request } from 'components/Request';
+import { TLSConfig } from 'components/TLSConfig';
+
+import { CheckIpVersion } from './CheckIpVersion';
+import { HttpCheckAuthentication } from './HttpCheckAuthentication';
+import { HttpCheckFollowRedirects } from './HttpCheckFollowRedirects';
+import { HttpCheckProxyURL } from './HttpCheckProxyURL';
+import { RequestBodyContentEncoding } from './RequestBodyContentEncoding';
+import { RequestBodyContentType } from './RequestBodyContentType';
+import { RequestBodyTextArea } from './RequestBodyTextArea';
+import { RequestHeaders } from './RequestHeaders';
+import { RequestQueryParams } from './RequestQueryParams';
+
+interface HttpRequestProps {
+  disabled?: boolean;
+  fields: HttpRequestFields;
+  onTest?: () => void;
+}
+
+export const HttpRequest = ({ disabled, fields, onTest }: HttpRequestProps) => {
+  const [showQueryParams, setShowQueryParams] = useState(false);
+  const { control, setValue, watch } = useFormContext();
+  const id = `request-method-${fields.method.name}`;
+  const styles = useStyles2(getStyles);
+  const theme = useTheme2();
+  const targetValue = watch(fields.target.name) as string;
+  const parsedURL = parseUrl(targetValue);
+
+  return (
+    <Request>
+      <Request.Field
+        data-fs-element="Check request target select"
+        description={`Full URL to send requests to`}
+        disabled={disabled}
+        htmlFor={id}
+        name={fields.target.name}
+      >
+        <div className={styles.grid}>
+          <Controller
+            control={control}
+            render={({ field }) => {
+              const { ref, onChange, ...rest } = field;
+              const value = field.value as HttpMethod;
+
+              return (
+                <div>
+                  <Select
+                    {...rest}
+                    aria-label={fields.method['aria-label'] || `Request method *`}
+                    className={css({
+                      borderColor: getMethodColor(theme, value),
+                    })}
+                    disabled={disabled}
+                    onChange={({ value }) => onChange(value)}
+                    options={METHOD_OPTIONS}
+                    tabSelectsValue={false}
+                  />
+                </div>
+              );
+            }}
+            name={fields.method.name}
+          />
+          <Request.Input
+            aria-label={fields.target['aria-label'] || `Request target *`}
+            data-fs-element="Target input"
```

Comment:
These two props seem to already be set in `RequestInput` by default, so it looks like they could be removed from here.